x/assets/keeper/genesis.go

```diff
@@ -40,21 +41,55 @@ func (k Keeper) InitGenesis(ctx sdk.Context, data *types.GenesisState) {
 			if err := k.UpdateStakerAssetState(
 				ctx, stakerID, assetID, infoAsChange,
 			); err != nil {
-				panic(err)
+				panic(errorsmod.Wrap(err, "failed to set deposit info"))
 			}
-			// now for the asset, increase the deposit value
-			if err := k.UpdateStakingAssetTotalAmount(
-				ctx, assetID, info.TotalDepositAmount,
-			); err != nil {
-				panic(err)
+		}
+	}
+
+	for _, assets := range data.OperatorAssets {
+		for _, assetInfo := range assets.AssetsState {
+			// #nosec G703 // already validated
+			accAddress, _ := sdk.AccAddressFromBech32(assets.Operator)
+			infoAsChange := types.DeltaOperatorSingleAsset(assetInfo.Info)
+			err := k.UpdateOperatorAssetState(ctx, accAddress, assetInfo.AssetID, infoAsChange)
+			if err != nil {
+				panic(errorsmod.Wrap(err, "failed to set operator asset info"))
 			}
 		}
 	}
 }
 
 // ExportGenesis returns the module's exported genesis.
-func (Keeper) ExportGenesis(sdk.Context) *types.GenesisState {
+func (k Keeper) ExportGenesis(ctx sdk.Context) *types.GenesisState {
 	res := types.GenesisState{}
-	// TODO
+	params, err := k.GetParams(ctx)
```

Comment:
why not use a single-line **if** statement to handle errors, assigning the value to 'res' upon **successfully retrieving data** ?

Reply:
Because it returns a pointer type. As for the other fields, I will change them to a single-line according to your suggestion.